src/components/organisms/AssetActions/Edit/index.tsx

```diff
@@ -76,6 +82,26 @@ export default function Edit({
         description: values.description
       })
 
+      console.log('herrrreeee')
+
+      const exchandeId = await ocean.fixedRateExchange.generateExchangeId(
```

Comment:
this is generating a new exchange ID which ocean.js uses internally. In here, we need to get the exchange ID based on the given datatoken address somehow.

Bummer that [`getExchange()`](https://github.com/oceanprotocol/ocean.js/blob/main/src/exchange/FixedRateExchange.ts#L319) requires an exchange ID 😄 

But I'm almost sure `ddo.price.address` is the exchange ID, it's the address of that specific fixed rate exchange contract

Reply:
but good exercise in error reporting. If it is the case that you tried to set the price of a randomly generated `exchangeId` (which does not exist) the transaction should have failed.

Reply:
well It wasn't really on purpose, I mean I wasn't sure if that is right or not :)